src/octoprint/templates/index.jinja2

```diff
@@ -97,7 +97,7 @@
                                     >
                                 <div class="accordion-heading">
                                     <a class="accordion-toggle" data-toggle="collapse" data-test-id="sidebar-{{ data._div }}-toggle" data-target="#{{ data._div }}">
-                                        {% if "icon" in data %}<i class="fa icon-black fa-{{ data.icon }}"></i> {% endif %}{{ entry|e }}
+                                        {% if "icon" in data %}<i class="fa icon-black fa-{{ data.icon }}"></i> {% endif %}{{ entry|e }} <!-- TODO fontawesom 5 upgrade - note this may not be possible for back-compat reasons - for 2.0.0? -->
```

Comment:
One thing I forgot to mention - I didn't change this line, in case plugins were using fa-4 icons, as it would break the compatibility shims (they are `fa.broken-icon`, so `fas.broken-icon` is not picked up). So no fa-5 icons in sidebar headers.